registry/storage/driver/hdfsweb/fs.go

```
@@ -0,0 +1,554 @@
+	u, err = url.ParseRequestURI(loc)
+	if err != nil {
+		return false, fmt.Errorf("Append(%s) - did not receive a valid URL from server.", loc)
+	}
+
+	req, _ = http.NewRequest("POST", u.String(), data)
+	rsp, err = client.Do(req)
+	if err != nil {
+		return false, err
+	}
+
+	if rsp.StatusCode != http.StatusOK {
+		defer rsp.Body.Close()
+		_, err = responseToHdfsData(rsp)
+		if err != nil {
+			return false, err
+		}
+		return false, fmt.Errorf("Append(%s) - File not created.  Server returned status %v", loc, rsp.StatusCode)
+	}
+
+	return true, nil
+}
+
+// Concat concatenate (on the server) a list of given files paths to a new file.
+func (fs *FileSystem) Concat(target string, sources []string) (bool, error) {
+	client := &http.Client{}
+	if target == "" {
+		return false, fmt.Errorf("Concat() - The target path must be provided.")
+	}
+	params := map[string]string{
+		"op":      "CONCAT",
+		"sources": strings.Join(sources, ","),
+	}
+
+	u, err := fs.buildRequestURL(target, &params)
+	if err != nil {
+		return false, err
+	}
+
+	req, _ := http.NewRequest("POST", u.String(), nil)
+	rsp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	if rsp.StatusCode != http.StatusOK {
+		defer rsp.Body.Close()
+		_, err = responseToHdfsData(rsp)
+		if err != nil {
+			return false, err
+		}
+		return false, fmt.Errorf("Concat(%s) - File not concatenated.  Server returned status %v", u.String(), rsp.StatusCode)
+	}
+	return true, nil
+}
+
+// Rename renames the specified path resource to a new name.
+// See HDFS FileSystem.rename()
+func (fs *FileSystem) Rename(source, destination string) (bool, error) {
+	if source == "" || destination == "" {
+		return false, fmt.Errorf("Rename() - params source and destination cannot be empty.")
+	}
+
+	params := map[string]string{"op": "RENAME", "destination": destination}
+	u, err := fs.buildRequestURL(source, &params)
+	if err != nil {
+		return false, err
+	}
+
+	req, _ := http.NewRequest("PUT", u.String(), nil)
```

Never ignore errors.